lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1423,6 +1405,36 @@ def write_store(
         )
         return rval
 
+    @router.delete(
+        "/api/workflows/{workflow_id}",
+        summary="Add the deleted flag to a workflow.",
+    )
+    def delete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        workflow_to_delete = self.workflows_manager.get_stored_workflow(trans, workflow_id)
+        if workflow_to_delete.user != trans.user and not trans.user_is_admin:
+            raise exceptions.InsufficientPermissionsException()
+        self.workflows_manager.delete(workflow_to_delete)
+        return self.workflow_contents_manager.workflow_to_dict(trans, workflow_to_delete, style="instance")
+
+    @router.post(
+        "/api/workflows/{workflow_id}/undelete",
+        summary="Remove the deleted flag from a workflow.",
+    )
+    def undelete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        workflow_to_undelete = self.workflows_manager.get_stored_workflow(trans, workflow_id)
```

Comment:
Ideally, the logic here will be in a method inside the service (which already has access to the WorkflowManager) and then you can return a 204 as Marius suggested:
```
self.service.undelete(trans, workflow_id)
return Response(status_code=status.HTTP_204_NO_CONTENT)
```
There is not much benefit to using just one service call here since this endpoint is simple enough, but it becomes more interesting when the endpoint logic is complex or involves multiple managers interacting, also it will be easier to migrate in the future to a new fancy API framework.